test/spec/modules/smartadserverBidAdapter_spec.js

```diff
@@ -1425,6 +1414,87 @@ describe('Smart bid adapter tests', function () {
       expect(bannerRequest).to.not.equal(null).and.to.not.be.undefined;
       expect(bannerRequest).to.have.property('bidfloor').and.to.equal(1.93);
     });
+
+    describe('#getBidFloor', () => {
+      let bid;
+      beforeEach(() => {
+        bid = {
+          mediaTypes: {
+            banner: {
+              sizes: [
+                [300, 250],
+                [300, 600]
+              ]
+            }
+          },
+          getFloor: (data) => {
+            if (data.mediaType === BANNER) {
+              if (data.size[0] === 300 && data.size[1] === 250) {
+                return { floor: 1.2 };
+              } else if (data.size[0] === 300 && data.size[1] === 600) {
+                return { floor: 1.4 };
+              } else if (data.size[0] === 30 && data.size[1] === 60) {
+                return 'string';
+              } else {
+                return { floor: 1.0 };
+              }
+            } else if (data.mediaType === VIDEO) {
+              if (data.size[0] === 640 && data.size[1] === 480) {
+                return { floor: 2.3 };
+              } else {
+                return { floor: 2.1 };
+              }
+            } else {
+              return {};
+            }
+          }
+        };
+      });
+
+      it('should return lowest floor from specified ones', () => {
+        expect(spec.getBidFloor(bid, 'DKK', BANNER)).to.deep.eq(1.2);
+      });
+
+      it('should return default floor for media type whatever size', () => {
+        bid.mediaTypes.banner.sizes.push([300, 400]);
+        expect(spec.getBidFloor(bid, 'DKK', BANNER)).to.deep.eq(1.0);
+      });
+
+      it('should return default floor', () => {
+        expect(spec.getBidFloor(bid, 'DKK', VIDEO)).to.deep.eq(0);
+      });
+
+      it('should return default floor when currency not passed', () => {
```

Comment:
For me this test is the same as first `it('should return lowest floor from specified ones')`
It's because the getFloor mock doesn't take into account currency. 
I guess when test says should return default then it should return 0 not 1.2, no?

Reply:
In this test I wanted to show that the currency is not mandatory and USD is taken if not passed. I agree that the tests are similar but not exactly the same. I could add the currency in the mock but since there is no conversion in the mock (which is already done before the floor module I understand) it's not needed. Maybe it will be more clear if I use USD instead of DKK everywhere.